services/web/client/source/class/qxapp/auth/ui/RegistrationPage.js

```diff
@@ -0,0 +1,120 @@
+
+/**
+ *
+ *  TODO: add a check to prevent bots to register users
+*/
+qx.Class.define("qxapp.auth.ui.RegistrationPage", {
+  extend: qxapp.auth.core.BaseAuthPage,
+
+
+  /*
+  *****************************************************************************
+     MEMBERS
+  *****************************************************************************
+  */
+
+  members: {
+
+    // overrides base
+    _buildPage: function() {
+      let manager = new qx.ui.form.validation.Manager();
+
+      this._addTitleHeader(this.tr("Registration"));
+
+      // email, pass1 == pass2
+      let email = new qx.ui.form.TextField();
+      email.setRequired(true);
+      email.setPlaceholder(this.tr("Introduce your email"));
+      this.add(email);
+
+      let uname = new qx.ui.form.TextField();
+      uname.setRequired(true);
+      uname.setPlaceholder(this.tr("Introduce a user name"));
+      this.add(uname);
+
+      let pass1 = new qx.ui.form.PasswordField();
+      pass1.setRequired(true);
+      pass1.setPlaceholder(this.tr("Introduce a password"));
+      this.add(pass1);
+
+      let pass2 = new qx.ui.form.PasswordField();
+      pass2.setRequired(true);
+      pass2.setPlaceholder(this.tr("Retype the password"));
+      this.add(pass2);
+
+      // interaction
+      email.addListener("changeValue", function(e) {
+        // Auto-guess
+        if (uname.getValue()=== null) {
+          let data = e.getData().split("@")[0];
+          uname.setValue(qx.lang.String.capitalize(qx.lang.String.clean(data)));
+        }
+      }, this);
+
+      // validation
+      manager.add(email, qx.util.Validate.email());
+      manager.add(pass1, function(value, itemForm) {
+        const isValid = value !== null && value.length > 2;
+        if (!isValid) {
+          itemForm.setInvalidMessage("Please enter a password at with least 3 characters.");
+        }
+        return isValid;
+      });
+      manager.setValidator(function(_itemForms) {
+        const isValid = pass1.getValue() == pass2.getValue();
+        if (!isValid) {
+          [pass1, pass2].forEach(pass => {
+            pass.setInvalidMessage("Passwords do not match.");
+            pass.setValid(isValid);
```

Comment:
```javascript
pass.set({
  invalidMessage: this.tr("Passwords do not match"),
  valid: false
});
```

Reply:
IN this case, ``this`` is not in the context of the function ... but can be solved using static member ``qx.locale.Manager.tr``